tests/storage/snapshot.go

```diff
@@ -436,14 +437,16 @@ var _ = SIGDescribe("VirtualMachineSnapshot Tests", func() {
 
 				//update vm to make sure vm revision is saved in the snapshot
 				By("Updating the VM template spec")
-				newVM, err := virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
+				err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+					updatedVM, err := virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &metav1.GetOptions{})
```

Comment:
Why is fetching the vm also inside the retry? Have you seen that this also fails occasionally?

Reply:
This is basically the pattern to follow in case of a collision.
You need to re-fetch the object, so it will include the latest resource version, change the data and then re-update. Repeating until succeeding or giving up at some point.